File: swea/D2/1940.가랏! RC카!/1940.py
T = int(input())
for tc in range(1, T+1):
    n = int(input())

    speed = 0
    dist = 0

    for _ in range(n):
        cmd = list(map(int, input().split()))

        #커맨드의 존재에 따라 속도를 추가하던가 말던가 하고 이동거리 추가
        if cmd[0] == 1:
            speed += cmd[1]
        
        elif cmd[0] == 2:
            speed -= cmd[1]
            
            if speed < 0:
                speed = 0
        
        dist += speed
    
    print(f"#{tc} {dist}")

File: swea/D2/1940.가랏! RC카!/1940.cpp
#include <iostream>

using namespace std;

int T;

int main (){
  cin >> T;

  for(int t = 1; t <= T; t++){
    int n;
    cin >> n;

    int speed = 0, dist = 0, cmd, acc;
    for(int i = 0; i < n; i++){
      cin >> cmd;
      if(cmd == 1){
        cin >> acc;
        speed += acc;
      }

      else if(cmd == 2){
        cin >> acc;
        speed -= acc;

        if(speed < 0)
          speed = 0;
      }

      dist += speed;
    }

    cout << '#' << t << ' ' << dist << endl;
  }
}